controllers/machine_controller.go

```diff
@@ -289,6 +308,39 @@ func (r *MachineReconciler) tryLock(machine *naglfarv1.Machine, dockerClient *do
 	return
 }
 
+func (r *MachineReconciler) createUploadDaemon(machine *naglfarv1.Machine, dockerClient *dockerutil.Client) (uploadPort int, err error) {
+	cfg, hostCfg := container.UploadDaemonCfg(string(machine.UID))
+	stat, err := dockerClient.Run(container.UploadDaemon, container.UploadDaemonImage, dockerutil.RunOptions{
+		Config:     cfg,
+		HostConfig: hostCfg,
+	})
+
+	if err != nil {
+		return
+	}
+
+	if stat.Config == nil || stat.Config.Labels == nil {
+		err = fmt.Errorf("invalid occupy container: %s", stat.Name)
+	}
+
+	uid, ok := stat.Config.Labels[container.UploadLabel]
+	if !ok {
+		err = fmt.Errorf("container %s has no label `%s`", stat.Name, container.UploadLabel)
+		return
+	}
+
+	if uid != string(machine.UID) {
+		err = fmt.Errorf("machine(%s) occupied by other naglfar system: UID(%s)", machine.Name, machine.UID)
+		return
+	}
+
+	if ports, ok := stat.NetworkSettings.Ports[(nat.Port)(strconv.Itoa(container.UploadDaemonInternalPort)+"/tcp")]; ok && len(ports) > 0 {
```

Comment:
Does the code only want to get the external Port? why not using container.UploadDaemonExternalPort?

Reply:
(nat.Port) is the type alias of string, please don't import a total module to only for importing a type alias!

Reply:
![image](https://user-images.githubusercontent.com/59600278/105653595-2aa01780-5ef7-11eb-9886-6ef7920a2b10.png)
In chaos daemon, port is string,I think integers are more reasonable.

Reply:
The externalPort is the port exposed by the host, and the internalPort is used between different containers

Reply:
ok.

Reply:
![image](https://user-images.githubusercontent.com/59600278/105664503-c9387280-5f0f-11eb-904d-6e9df3b252bf.png)
![image](https://user-images.githubusercontent.com/59600278/105664523-d48b9e00-5f0f-11eb-8184-27ec1a339355.png)
Is not a constant, can not be used directly, need type conversion

Reply:
done